### PR TITLE
Add Java Annotations to provide better information for serialising Rosetta Objects

### PR DESCRIPTION
**Pull request by minesh-s-patel**
See https://github.com/REGnosys/rosetta-dsl/issues/613

### CODE REVIEW

File: rosetta-testing/src/test/java/com/regnosys/rosetta/generator/java/object/ModelObjectGeneratorTest.xtend
Comment (SimonCockx) [inline rosetta-testing/src/test/java/com/regnosys/rosetta/generator/java/object/ModelObjectGeneratorTest.xtend:1]: File can be reverted.

File: rosetta-runtime/src/main/java/com/rosetta/model/lib/annotations/RosettaDataType.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+@Documented
+@Retention(RetentionPolicy.RUNTIME)
+public @interface RosettaDataType {
+
+	String value() default "";
+	
+	Class<? extends RosettaModelObjectBuilder> builder();
+	
+	String version() default "";
```
Comment (SimonCockx) [inline rosetta-runtime/src/main/java/com/rosetta/model/lib/annotations/RosettaDataType.java:17]: Got a question about this one. See Slack.

**Review by SimonCockx — APPROVED** [review]

File: rosetta-runtime/src/main/java/com/rosetta/model/lib/annotations/RosettaDataType.java (hunk 1 above)
Comment (minesh-s-patel) [reply rosetta-runtime/src/main/java/com/rosetta/model/lib/annotations/RosettaDataType.java:17]: I thought about this and I think its OK to keep this on all files that are generated.

File: rosetta-testing/com.regnosys.rosetta.tests.iml
[the remark this answers is not in the document]
Comment (minesh-s-patel) [reply rosetta-testing/com.regnosys.rosetta.tests.iml:1]: Done

File: rosetta-testing/src/test/java/com/regnosys/rosetta/generator/java/object/ModelObjectGeneratorTest.xtend
Comment (minesh-s-patel) [reply rosetta-testing/src/test/java/com/regnosys/rosetta/generator/java/object/ModelObjectGeneratorTest.xtend:1]: Done

**Review by hugohills-regnosys — APPROVED** [review]

**Review by SimonCockx — APPROVED** [review]

**Pull request merged**